lib/jsdom/living/nodes/Document-impl.js

```diff
@@ -140,6 +282,128 @@ class RequestManager {
   }
 }
 
+class PerDocumentResourceLoader {
+  constructor(document) {
+    this._document = document;
+    this._defaultEncoding = document._encoding;
+    this._resourceLoader = document._defaultView ? document._defaultView._resourceLoader : new ResourceLoader();
+    this._requestManager = document._requestManager;
+    this._queue = document._queue;
+    this._deferQueue = document._deferQueue;
+    this._asyncQueue = document._asyncQueue;
+
+    // TODO: Remove when _hasFeature is no longer necessary
```

Comment:
If you want you can remove all the _hasFeature stuff yourself in this PR :)

Reply:
If I remove that, I need to remove the old api too, right?